combo_cases_and_deaths/delphi_combo_cases_and_deaths/run.py

```diff
@@ -107,25 +141,56 @@ def combine_usafacts_and_jhu(signal, geo, date_range, issue_range=None, fetcher=
         geo_to_fetch,
         issues=issue_range
     )
+
     if check_none_data_frame(usafacts_df, "USA-FACTS", date_range) and \
-       (geo_to_fetch not in ('state', 'county') or
+       (geo_to_fetch not in ('state', 'county', 'msa') or
         check_none_data_frame(jhu_df, "JHU", date_range)):
+        return None
+
+    merged_df = merge_dfs_by_geos(usafacts_df, jhu_df, geo_to_fetch)
+    unique_dates = merged_df["timestamp"].unique()
+    return unique_dates
+
+
+def combine_usafacts_and_jhu(signal, geo, date_range, issue_range=None, fetcher=covidcast.signal):
+    """Add rows for PR from JHU signals to USA-FACTS signals.
+
+    For hhs and nation, fetch the county `num` data so we can compute the proportions correctly
+    and after combining JHU and USAFacts and mapping to the desired geos.
+    """
+    is_special_geo = geo in ["hhs", "nation"]
+    geo_to_fetch = "county" if is_special_geo else geo
+    signal_to_fetch = signal.replace("_prop", "_num") if is_special_geo else signal
+
+    unique_dates = get_updated_dates(signal, geo, date_range, issue_range, fetcher)
+    # This occurs if the usafacts ~and the jhu query were empty
+    if unique_dates is None:
         return pd.DataFrame({}, columns=COLUMN_MAPPING.values())
+    # Requery each date with as_of=today, so that every geo is represented
+    query_date_pairs = [(date, date) for date in unique_dates]
```

Comment:
this is probably fine but we should check the running time; worst case scenario this would generate 365 calls to fetcher instead of 1 if all dates in the past year were touched by some issue

Reply:
the runtime is definitely a concern. I will test something like 100 calls to fetcher for 100 separate dates vs 1 call to fetcher for a range of 100 dates.

Reply:
(it will matter what geo you use there -- fetcher will automatically split county queries into one API call per day, but it bundles state/hhs/nation queries in bigger batches)

Reply:
oof, ok so the new method of calling fetcher once for each date seems to be 10x slower than just calling a whole range. for both state and county geos. the performance gap shrinks a little when there's sparsity in `unique_dates`, but even for `date_range = pd.date_range("2020-08-1", "2021-03-1", freq='W')`, fetching the whole range in a single call is ~4 times faster than querying those individual days.

so i think it makes more sense to do a single fetch call to `unique_dates.min(), unique_dates.max()` instead.